scripts/linters/pylint_extensions.py

```diff
@@ -2109,6 +2111,44 @@ def visit_call(self, node):
                     break
 
 
+class StringConcatenationChecker(checkers.BaseChecker):
+    """Custom pylint checker which prohibits use of string concatenation and
+     promotes string interpolation instead.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'string-concatenation'
+    priority = -1
+    msgs = {
+        'C0034': (
+            'at string %s, use string interpolation (\'string1%%s\' %% string2)'
+            ' rather than string concatenation (\'string1\' + \'string2\').',
+            'string-concatenation',
+            'Enforce use of string interpolation over string concatenation.',
+        ),
+    }
+
+    def visit_binop(self, node):
+        """Called for every '+' operator to prohibit usage of string
+         concatenation. It covers only those cases where at least one of the
+         operands is astroid.nodes.Const.
+
+        Args:
+            node: astroid.node.BinOp. Node to access module content.
+        """
+        if node.op == b'+':
```

Comment:
In general try to return in a function as soon as possible. Like replacing the `if` condition with 
```python
if node.op != b'+':
    return
```
and then the rest of the code follows. This also saves us from extra rightward indentation in the rest of the code, improving readability.

Reply:
Yes, I really liked the idea. Done!!

Reply:
Thanks for the suggestion!!!  Done